platform/mk2/hal/CAN_stm32/CAN_device_stm32.c

```diff
@@ -189,57 +184,76 @@ static void CAN_device_init_2(int baud)
 
 int CAN_device_init(uint8_t channel, uint32_t baud)
 {
-
-    pr_info("CAN");
-    pr_info_int(channel);
-    pr_info(" init @ ");
-    pr_info_int(baud);
-
-    if (initQueues()) {
-        switch (channel) {
-        case 0:
-            CAN_device_init_1(baud);
-            break;
-        case 1:
-            CAN_device_init_2(baud);
-            break;
-        }
-        pr_info(" win\r\n");
-        return 1;
-    } else {
-        pr_info(" fail\r\n");
-        return 0;
-    }
+	pr_info("Initializing CAN");
+	pr_info_int(channel);
+	pr_info_int_msg(" with baud rate ", baud);
```

Comment:
Yep.  Its very close.  Currently on float and double printing fail.  Gotta figure out why.  Beyond that we will have it.